trie-db/src/trie_codec.rs

```diff
@@ -398,8 +409,8 @@ pub fn decode_compact<L, DB, T>(db: &mut DB, encoded: &[Vec<u8>])
 	// The prefix of the next item to be read from the slice of encoded items.
 	let mut prefix = NibbleVec::new();
 
-	for (i, encoded_node) in encoded.iter().enumerate() {
-		let node = L::Codec::decode(encoded_node)
+	for (i, mut encoded_node) in encoded.enumerate() {
+		let node = L::Codec::decode(&mut encoded_node)
```

Comment:
why is the mutable reference needed? as far as I can see, `decode` accepts an immutable reference

Reply:
:+1: , I confuse the trait with the parity codec trait where it would have been &Input with implementatiion of input being '&mut &[u8]', but here it is  clearly not needed.